applications/GeoMechanicsApplication/tests/cpp_tests/test_stress_strain_utitlities.cpp

```diff
@@ -0,0 +1,115 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Wijtze Pieter Kikstra
+//
+
+#include "custom_utilities/math_utilities.hpp"
+#include "custom_utilities/stress_strain_utilities.h"
+#include "testing/testing.h"
+#include <boost/numeric/ublas/assignment.hpp>
+
+using namespace Kratos;
+
+namespace Kratos::Testing
+{
+
+KRATOS_TEST_CASE_IN_SUITE(CheckCalculateTrace, KratosGeoMechanicsFastSuite)
+{
+    Vector eye = MathUtils<double>::StressTensorToVector(IdentityMatrix(3));
+    KRATOS_EXPECT_DOUBLE_EQ(3.0, StressStrainUtilities::CalculateTrace(eye));
+}
+
+KRATOS_TEST_CASE_IN_SUITE(CheckCalculateMeanStress, KratosGeoMechanicsFastSuite)
+{
+    Vector eye = MathUtils<double>::StressTensorToVector(IdentityMatrix(3));
+    KRATOS_EXPECT_DOUBLE_EQ(1.0, StressStrainUtilities::CalculateMeanStress(eye));
+}
+
+KRATOS_TEST_CASE_IN_SUITE(CheckGreenLagrangeStrainTensor, KratosGeoMechanicsFastSuite)
+{
+    Matrix eye = StressStrainUtilities::CalculateGreenLagrangeStrainTensor(sqrt(3.) * IdentityMatrix(3));
+    KRATOS_EXPECT_DOUBLE_EQ(1.0, eye(0, 0));
+    KRATOS_EXPECT_DOUBLE_EQ(1.0, eye(1, 1));
+    KRATOS_EXPECT_DOUBLE_EQ(1.0, eye(2, 2));
+    KRATOS_EXPECT_DOUBLE_EQ(0.0, eye(0, 1));
+    KRATOS_EXPECT_DOUBLE_EQ(0.0, eye(0, 2));
+    KRATOS_EXPECT_DOUBLE_EQ(0.0, eye(1, 2));
```

Comment:
Wouldn't it be more elegant if we'd use `KRATOS_EXPECT_MATRIX_NEAR` to check the returned matrix?

Reply:
yes indeed.